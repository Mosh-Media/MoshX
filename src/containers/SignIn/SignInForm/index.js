import React, { useState } from "react";

import * as ROUTES from "../../../constants/routes";
import { compose } from "recompose";
import EmailSignInForm from "../../../components/EmailSignInForm";

import { withFirebase } from "../../../components/Firebase";
import { withRouter } from "react-router-dom";

const INITIAL_STATE = {
	email: "",
	password: "",
	error: null
};

const SignInFormBase = ({ firebase, history}) => {
	const [state, setState] = useState({ ...INITIAL_STATE });
	const { email, password, error } = state;

	const handleSubmit = event => {
		firebase
			.signInWithEmailAndPassword(email, password)
			.then(() => {
				setState({ ...INITIAL_STATE });
				history.push(ROUTES.HOME.path);
			})
			.catch(error => {
				setState({ ...state, error });
			});

		event.preventDefault();
	};

	const onChange = event => {
		setState({ ...state, [event.target.name]: event.target.value });
	};

	return (
		<EmailSignInForm
			onSubmit={handleSubmit}
			isInvalid={password === "" || email === ""}
			onChange={onChange}
			error={error}
		/>
	);
};

const SignInForm = compose(withRouter, withFirebase)(SignInFormBase);

export default SignInForm;