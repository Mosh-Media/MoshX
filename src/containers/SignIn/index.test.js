import React from "react";

import SignInForm from "./index.js";
import ShallowRenderer from "react-test-renderer/shallow";
import useGlobal from "store";
import useGlobalHook from "use-global-hook";

beforeEach(() => {
  useGlobal.mockClear();
  useGlobalHook.mockClear();
});

afterEach(() => {
  jest.restoreAllMocks();
});

// Mock functionality of global and store hooks
jest.mock("store", () => jest.fn().mockReturnValue([]));
jest.mock("use-global-hook", () => jest.fn());

const renderer = new ShallowRenderer();

describe("SignInForm Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<SignInForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
