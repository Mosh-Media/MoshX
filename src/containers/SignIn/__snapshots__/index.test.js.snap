// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm Component should match the snapshot 1`] = `
<Context.Consumer>
  [Function]
</Context.Consumer>
`;
